### ISSUE TITLE
Response.text panics when using charsets feature

### ISSUE THREAD

**Issue by tdryer** [sbstp/attohttpc#32]
When the `charsets` feature is enabled, `Response.text` panics in this program:

```
fn main() -> Result<(), attohttpc::Error> {
    let resp = attohttpc::get("https://rust-lang.org/").send()?;
    println!("{}", resp.text()?); // panic here
    Ok(())
}
```

The panic does not occur if the `charsets` feature is disabled, or if `Response.text_utf8` is used instead.

Version:
```
attohttpc = {"version" = "0.8.0", "features" = ["charsets"]}
```

Backtrace:
```
thread 'main' panicked at 'slice index starts at 12288 but ends at 8192', src/libcore/slice/mod.rs:2670:5
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:61
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1028
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1412
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:65
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:50
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:188
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:205
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:464
  11: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:373
  12: rust_begin_unwind
             at src/libstd/panicking.rs:302
  13: core::panicking::panic_fmt
             at src/libcore/panicking.rs:139
  14: core::slice::slice_index_order_fail
             at src/libcore/slice/mod.rs:2670
  15: <core::ops::range::Range<usize> as core::slice::SliceIndex<[T]>>::index_mut
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libcore/slice/mod.rs:2857
  16: <core::ops::range::RangeFrom<usize> as core::slice::SliceIndex<[T]>>::index_mut
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libcore/slice/mod.rs:2933
  17: core::slice::<impl core::ops::index::IndexMut<I> for [T]>::index_mut
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libcore/slice/mod.rs:2657
  18: <alloc::vec::Vec<T> as core::ops::index::IndexMut<I>>::index_mut
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/liballoc/vec.rs:1873
  19: std::io::read_to_end_with_reservation
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/io/mod.rs:389
  20: std::io::read_to_end
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/io/mod.rs:356
  21: std::io::Read::read_to_string::{{closure}}
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/io/mod.rs:713
  22: std::io::append_to_string
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/io/mod.rs:333
  23: std::io::Read::read_to_string
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/io/mod.rs:713
  24: attohttpc::parsing::response_reader::ResponseReader::text_with
             at /home/tom/.cargo/registry/src/github.com-1ecc6299db9ec823/attohttpc-0.8.0/src/parsing/response_reader.rs:129
  25: attohttpc::parsing::response_reader::ResponseReader::text
             at /home/tom/.cargo/registry/src/github.com-1ecc6299db9ec823/attohttpc-0.8.0/src/parsing/response_reader.rs:117
  26: attohttpc::parsing::response::Response::text
             at /home/tom/.cargo/registry/src/github.com-1ecc6299db9ec823/attohttpc-0.8.0/src/parsing/response.rs:143
  27: attohttpc_example::main
             at src/main.rs:3
  28: std::rt::lang_start::{{closure}}
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/rt.rs:61
  29: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:48
  30: std::panicking::try::do_call
             at src/libstd/panicking.rs:287
  31: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:78
  32: std::panicking::try
             at src/libstd/panicking.rs:265
  33: std::panic::catch_unwind
             at src/libstd/panic.rs:396
  34: std::rt::lang_start_internal
             at src/libstd/rt.rs:47
  35: std::rt::lang_start
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libstd/rt.rs:61
  36: main
  37: __libc_start_main
  38: _start
```

**Comment by sbstp** [comment]
Hi! Thanks for the report. I'm checking this out right now. Will update shortly.

**Comment by sbstp** [comment]
@tdryer hey, this is fixed. It was released in version `0.9.0`. Thanks for the report!
